File: Difficulty: Medium/Indexes of Subarray Sum/indexes-of-subarray-sum.java
class Solution {
    static ArrayList<Integer> subarraySum(int[] arr, int target) {
        int start = 0;
        int currentSum = 0;

        for (int end = 0; end < arr.length; end++) {
            currentSum += arr[end];

            while (currentSum > target && start <= end) {
                currentSum -= arr[start];
                start++;
            }

            if (currentSum == target) {
                return new ArrayList<>(Arrays.asList(start + 1, end + 1)); // 1-based index
            }
        }

        return new ArrayList<>(Arrays.asList(-1));
    }
}

File: Difficulty: Medium/Indexes of Subarray Sum/indexes-of-subarray-sum.cpp
class Solution {
  public:
    vector<int> subarraySum(vector<int> &arr, int target) {
        // code here
        int n = arr.size();
        int start = 0;
        int curr_sum = 0;
        for(int i = 0 ; i < n ; i++)
        {
            curr_sum += arr[i];
            while(curr_sum > target && start <= i)
            {
                curr_sum -= arr[start];
                start++;
            }
            if(curr_sum == target)
            {
                return {start + 1 , i + 1}; 
            }
        }
        
        return {-1};
    }
};